### PR TITLE
feat: support proprties of objects

### LINKED ISSUE #207: Collection analyzers are not running on properties of types
### Description
The various collection analyzers don't work if the asserted variable is a property of a type.

### Complete minimal example reproducing the issue
```csharp
[Fact]
public void Test()
{
    List<int> list = new() { 1, 2 };
    var obj = new TestType();

    // Correctly shows analyzer.
    list.Any().Should().BeTrue();

    // Doesn't show analyzer.
    obj.List.Any().Should().BeTrue();
}

public class TestType
{
    public List<int> List { get; set; }
}
```

### Expected behavior:
Assertion on collection properties should also show the analyzer message.

### Actual behavior:
Assertion on collection properties do not show the analyzer message.

### Versions
* Which version of Fluent Assertions Analyzers are you using? 0.21.0
* Which .NET runtime and version are you targeting? .NET 7

### PR DESCRIPTION
**Pull request by Meir017**
resolves #207

### CODE REVIEW

File: src/FluentAssertions.Analyzers/Utilities/FluentAssertionsAnalyzer.cs [hunk 1]
```diff
@@ -62,9 +62,11 @@ protected virtual Diagnostic AnalyzeExpression(ExpressionSyntax expression, Sema
             var variableNameExtractor = new VariableNameExtractor(semanticModel);
             expression.Accept(variableNameExtractor);
 
-            if (variableNameExtractor.VariableIdentifierName == null) return null;
-            var typeInfo = semanticModel.GetTypeInfo(variableNameExtractor.VariableIdentifierName);
-            if (!ShouldAnalyzeVariableTypeCore(typeInfo.Type, semanticModel)) return null;
+            if (variableNameExtractor.PropertiesAccessed
```
Comment (Meir017) [inline src/FluentAssertions.Analyzers/Utilities/FluentAssertionsAnalyzer.cs:65]: need to add more tests for edge cases where some nested properties are not really interesting (specifically `Count` and a few more)
